tss-esapi/src/abstraction/hashing.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::interface_types::algorithm::HashingAlgorithm;
+
+/// Provides the value of the digest used in this crate for the digest.
+pub trait AssociatedHashingAlgorithm {
+    /// Value of the digest when interacting with the TPM.
+    const TPM_DIGEST: HashingAlgorithm;
+}
+
+#[cfg(feature = "sha1")]
```

Comment:
Hmm... how does this work if I don't see the feature declared in Cargo.toml? :thinking:

Reply:
yeaaah, I learned this one recently too lol.
https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies
```
By default, this optional dependency implicitly defines a feature that looks like this:

[features]
gif = ["dep:gif"]
```

I'm abusing that mechanism here.

I actually need to split the features in the `Cargo.toml` to look more like:
```
rustcrypto = ["ecdsa", "elliptic-curve", "signature", "x509-cert"]
rustcrypto-full = ["rustcrypto", "p192", "p224", "p256", "p384", "p521", "rsa", "sha1", "sha2", "sha3", "sm2", "sm3"]
```

That way you can bring in rustcrypto + "p256" + "sha2" if you only care about those.